### PR TITLE
Populate joblist table for gateway jobs

### PR DESCRIPTION
**Pull request by jsperhac**
## Description
Provide back-end configuration needed for job performance drilldown and data export for the Gateways realm. The schema and ETL changes enabled by this configuration are:
- Break out gateways jobs aggregation by day from that by month, quarter, year
- Populate day aggregation job_id_list, last_modified, and id columns for gateways jobs
- Create and populate jobfact_by_day_joblist table for gateways
- Incorporate Gateways in rawstatistics configuration

Query classes have been adapted from those that govern the Jobs realm to enable display of Gateway jobs data in the job viewer. (These should be subclassed from existing Query classes later, once the Jobs Query classes are appropriately generalized/refactored, see also PR #1473) 

Note that bootstrapping and automatic testing is not included in the final version of this code, so that this schema and realm are not created by default in an installation. Therefore, component and unit test artifacts that pertain to multiple realms, including this one, have been backed out here. Instructions for the installation of this realm will be provided separately (see PR #1509).

## Motivation and Context

This PR extends and enhances the back-end Gateways realm support. The configuration provided here makes schema changes that enable direct drilldown to performance records from gateways jobs; and enable batch export of raw data from the Gateways realm (where this access is permitted by ACLs). 

This PR also fixes an aggregation bug (unreported, thus no link here) that was present in the original gateways back-end configuration (PR #1262).

Note that the configuration changes as provided here must be run manually via the ETL v2 pipeline to provide the schema changes (bootstrap; ingestion; aggregation).
## Tests performed

- Developed/deployed in a local Docker; ran all etl v2 governing gateways schema bootstrapping, ingestion, aggregation using small reference dataset. 
- Ran etl v2 on staging instance to verify/identify aggregation bug. 
- Ran all tests (adapted test artifacts to accommodate these new configurations).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: configuration/rawstatistics.d/30_gateways.json [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+                "group": "Requested Resource",
+                "documentation": "HIERARCHY_MIDDLE_LEVEL_INFO",
+                "batchExport": true
+            },
+            {
+                "name": "HIERARCHY_BOTTOM_LEVEL_LABEL",
+                "tableAlias": "fos",
+                "column": "description",
+                "group": "Requested Resource",
+                "documentation": "HIERARCHY_BOTTOM_LEVEL_INFO",
```
Comment (jtpalmer) [inline configuration/rawstatistics.d/30_gateways.json:358]: Just as an FYI for now, I'm working on changing the `HIERARCHY_` substitutions to match the style used by the ETL configuration files (See #1448).

File: configuration/rawstatistics.d/30_gateways.json [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+            {
+                "name": "Queue",
+                "tableAlias": "jr",
+                "column": "queue",
+                "group": "Requested Resource",
+                "documentation": "The name of the queue to which the job was submitted.",
+                "batchExport": true
+            },
+            {
+                "name": "HIERARCHY_TOP_LEVEL_LABEL",
```
Comment (jtpalmer) [inline configuration/rawstatistics.d/30_gateways.json:338]:
```suggestion
                "name": "${HIERARCHY_TOP_LEVEL_LABEL}",
```

File: configuration/rawstatistics.d/30_gateways.json [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+                "tableAlias": "fos",
+                "column": "directorate_description",
+                "group": "Requested Resource",
+                "documentation": "HIERARCHY_TOP_LEVEL_INFO",
```
Comment (jtpalmer) [inline configuration/rawstatistics.d/30_gateways.json:342]:
```suggestion
                "documentation": "${HIERARCHY_TOP_LEVEL_INFO}",
```

File: configuration/rawstatistics.d/30_gateways.json [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+                "batchExport": true
+            },
+            {
+                "name": "HIERARCHY_MIDDLE_LEVEL_LABEL",
```
Comment (jtpalmer) [inline configuration/rawstatistics.d/30_gateways.json:346]:
```suggestion
                "name": "${HIERARCHY_MIDDLE_LEVEL_LABEL}",
```

File: configuration/rawstatistics.d/30_gateways.json [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+                "tableAlias": "fos",
+                "column": "parent_description",
+                "group": "Requested Resource",
+                "documentation": "HIERARCHY_MIDDLE_LEVEL_INFO",
```
Comment (jtpalmer) [inline configuration/rawstatistics.d/30_gateways.json:350]:
```suggestion
                "documentation": "${HIERARCHY_MIDDLE_LEVEL_INFO}",
```

File: configuration/rawstatistics.d/30_gateways.json [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+                "batchExport": true
+            },
+            {
+                "name": "HIERARCHY_BOTTOM_LEVEL_LABEL",
```
Comment (jtpalmer) [inline configuration/rawstatistics.d/30_gateways.json:354]:
```suggestion
                "name": "${HIERARCHY_BOTTOM_LEVEL_LABEL}",
```

File: configuration/rawstatistics.d/30_gateways.json (hunk 1 above)
Comment (jtpalmer) [inline configuration/rawstatistics.d/30_gateways.json:358]:
```suggestion
                "documentation": "${HIERARCHY_BOTTOM_LEVEL_INFO}",
```

**Review by jtpalmer — CHANGES_REQUESTED** [review]
I merged in those variable substitution changes, here's what should change.

File: configuration/rawstatistics.d/30_gateways.json (hunk 1 above)
Comment (jsperhac) [reply configuration/rawstatistics.d/30_gateways.json]: The HIERARCHY_ substitutions for the rawstatistics.d file have been made as suggested.

**Comment by jsperhac** [comment]
@jtpalmer, I have a question about the component tests for data export. I see test failures that seem to be expecting array elements with [specific numeric keys](https://app.shippable.com/github/ubccr/xdmod/runs/6466/1/console). 

Updates to the component tests for the new realm don't help, since there are gaps in the keys; see my [most recent commit](https://github.com/ubccr/xdmod/pull/1411/commits/beed800cb8abbdceb7ed8413468d35a3c5644e6d). Am I missing something, or do we need an accommodation similar to the RawStatisticsConfiguration getRawDataRealms() method?

**Comment by jtpalmer** [comment]
I think the test failures are due to more of my `array_filter` bugs.  I created #1481 to fix that.

**Comment by jsperhac** [comment]
> I merged in those variable substitution changes, here's what should change.

These have been addressed.

**Review by jtpalmer — APPROVED** [review]

**Review by jtpalmer — APPROVED** [review]

**Pull request merged**